server/routes/nl.py

```diff
@@ -255,38 +271,41 @@ def _detection(orig_query, cleaned_query) -> Detection:
   model = current_app.config['NL_MODEL']
 
   # Step 1: find all relevant places and the name/type of the main place found.
-  places_found = model.detect_place(cleaned_query)
+  places_str_found = model.detect_place(cleaned_query)
 
-  if not places_found:
+  if not places_str_found:
     logging.info("Place detection failed.")
 
-  logging.info("Found places: {}".format(places_found))
-  # If place_dcid was already set by the url, skip inferring it.
-  place_dcid = request.args.get('place_dcid', '')
-  if not place_dcid and places_found:
-    place_dcid = _infer_place_dcid(places_found)
+  logging.info("Found places in query: {}".format(places_str_found))
+
+  query = cleaned_query
+  place_dcids = []
+  main_place = None
+  resolved_places = []
 
-  if place_dcid:
-    place_types = dc.property_values([place_dcid], 'typeOf')[place_dcid]
-    main_place_type = _get_preferred_type(place_types)
-    main_place_name = dc.property_values([place_dcid], 'name')[place_dcid][0]
+  # Look to find place DCIDs.
+  if places_str_found:
+    place_dcids = _infer_place_dcids(places_str_found)
+    logging.info(f"Found {len(place_dcids)} place dcids: {place_dcids}.")
 
+  for place_dcid in place_dcids:
+    # Set the Place objects (dcid, name, type).
+    resolved_places.append(_get_place_from_dcid(place_dcid))
+    logging.info(
+        f"Resolved {len(resolved_places)} place dcids: {resolved_places}.")
```

Comment:
Might be nice to pass all DCIDs in the DC calls?  (so sort of cut serial RPCs / BT round-trips)

Reply:
Done. PTAL